src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/TagHelperDescriptorJsonConverter.cs

```diff
@@ -828,7 +834,8 @@ private static void ReadDiagnostic(JsonReader reader, RazorDiagnosticCollection
                 }
             });
 
-            var descriptor = new RazorDiagnosticDescriptor(id, () => message, (RazorDiagnosticSeverity)severity);
+            // TODO: Needed?
```

Comment:
Yes; however, your lambda is capturing the "message" parameter which means that string will be held in memory to re-evaluate the cached method with it. Get the cached parameter ahead of time and use that in the descriptor lambda so the original `message` reference can be gc'd

Reply:
We talked about this offline and ended up landing on this being dealt with in another PR (the amount of memory being captured here was actually pretty big so it's a nice catch and relatively easy savings).

Reply:
You'll still want to calculate the message beforehand here. Even with the other PR we'll want that 😄